### DISCUSSION TITLE
Make 'relax' mod more intelligent

### DISCUSSION THREAD

**Discussion by nana123126** [ppy/osu#24871] [Ideas]
For games with relax mod, it will result in the inability to click early, which will result in a large number of notes in some players' Misses because they often move the mouse out of the circle early

So the functions that should be enhanced are:
1. Make it click when the mouse is about to leave the circle, unless clicking will cause a miss or the object to be clicked is not the earliest existing note (excluding the slider that is being held down).

If it will cause a balance impact, you can try to further increase the penalty for low acc to prevent random movement and reduce aim pp at low OD.

**Comment by Rekunan** [comment]
> If it will cause a balance impact, you can try to further increase the penalty for low acc to prevent random movement and reduce aim pp at low OD.

FWIW, the current relax pp implementation is to count all acc drops as misses, so this is technically already built-in.

**Reply by nana123126** [reply]
Um I don't quite understand the meaning of this sentence.

If you mean that the acc pp has been removed, I know that there is no accuracy score brought by the acc pp, but there is still aim pp, which should have higher penalties for low accuracy, which means it is also related to accuracy.

If you mean that aim app has already judged all click results except for 'great' as' miss' when calculating, then I don't think this is good because it means that the scores except for 'great' evaluations are very poor, and they will have no difference (except for combo, but 'MEH' and 'OK' are still the same). Although it does solve the problem of random movement.

**Reply by Rekunan** [reply]
I mean the second.

I agree there should be more done, but it works as a quick basic implementation for now (tap pp is also removed).
